Assets/Folder.php

```diff
@@ -194,14 +194,11 @@ public function getCMSFields() {
 		// Don't show readonly path until we can implement parent folder selection,
 		// it's too confusing when readonly (makes sense for files only).
 
-		$fields = FieldList::create([
-			HeaderField::create('TitleHeader', $this->Title, 1),
-			LiteralField::create("ImageFull", $this->PreviewThumbnail()),
-			TextField::create("Name", $this->fieldLabel('Filename')),
-			HiddenField::create('ID', $this->ID)
-		]);
-
-		$this->extend('updateCMSFields', $fields);
```

Comment:
This changes the API unnecessarily - 3.x was creating a fieldset from scratch: https://github.com/silverstripe/silverstripe-framework/blob/3.4/filesystem/Folder.php#L461

So if we're using `parent::getCMSFields()` here, assumptions made in `File->updateCMSFields()` on 3.x installs will break: It'll add fields to `Folder` which were only intended for `File`.